lib/gpu/src/context.rs

```diff
@@ -112,6 +112,45 @@ impl Context {
         Ok(())
     }
 
+    pub fn barrier_buffers(&mut self, buffers: &[Arc<Buffer>]) -> GpuResult<()> {
+        if self.vk_command_buffer == vk::CommandBuffer::null() {
+            self.init_command_buffer()?;
+        }
+
+        let buffer_memory_barriers = buffers
+            .iter()
+            .map(|buffer| {
+                self.resources.push(buffer.clone() as Arc<dyn Resource>);
```

Comment:
It seems that each call to `barrier_buffers` would extend the `self.resources` vec.